kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java

```diff
@@ -0,0 +1,375 @@
+          }
+          return LongType.get();
+        case FLOAT32:
+          return FloatType.get();
+        case FLOAT64:
+          return DoubleType.get();
+        case ARRAY:
+          Type elementType = toIcebergType(valueSchema.valueSchema());
+          if (config.schemaForceOptional() || valueSchema.valueSchema().isOptional()) {
+            return ListType.ofOptional(nextId(), elementType);
+          } else {
+            return ListType.ofRequired(nextId(), elementType);
+          }
+        case MAP:
+          Type keyType = toIcebergType(valueSchema.keySchema());
+          Type valueType = toIcebergType(valueSchema.valueSchema());
+          if (config.schemaForceOptional() || valueSchema.valueSchema().isOptional()) {
+            return MapType.ofOptional(nextId(), nextId(), keyType, valueType);
+          } else {
+            return MapType.ofRequired(nextId(), nextId(), keyType, valueType);
+          }
+        case STRUCT:
+          List<NestedField> structFields =
+              valueSchema.fields().stream()
+                  .map(
+                      field ->
+                          NestedField.of(
+                              nextId(),
+                              config.schemaForceOptional() || field.schema().isOptional(),
+                              field.name(),
+                              toIcebergType(field.schema())))
+                  .collect(toList());
+          return StructType.of(structFields);
+        case STRING:
+        default:
+          return StringType.get();
+      }
+    }
+
+    Optional<Type> inferIcebergType(Object value) {
+      return Optional.ofNullable(doInferIcebergType(value));
+    }
+
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    private Type doInferIcebergType(Object value) {
+      if (value == null) {
+        return null;
+      } else if (value instanceof String) {
+        return StringType.get();
+      } else if (value instanceof Boolean) {
+        return BooleanType.get();
+      } else if (value instanceof BigDecimal) {
+        BigDecimal bigDecimal = (BigDecimal) value;
+        return DecimalType.of(bigDecimal.precision(), bigDecimal.scale());
+      } else if (value instanceof Number) {
+        Number num = (Number) value;
+        Double dbl = num.doubleValue();
+        if (dbl.equals(Math.floor(dbl))) {
+          return LongType.get();
+        } else {
+          return DoubleType.get();
+        }
+      } else if (value instanceof LocalDate) {
+        return DateType.get();
+      } else if (value instanceof LocalTime) {
+        return TimeType.get();
+      } else if (value instanceof java.util.Date || value instanceof OffsetDateTime) {
+        return TimestampType.withZone();
+      } else if (value instanceof LocalDateTime) {
+        return TimestampType.withoutZone();
+      } else if (value instanceof List) {
+        List<?> list = (List<?>) value;
+        if (list.isEmpty()) {
+          return null;
```

Comment:
of strings?

Reply:
I felt it was better to skip adding it if we can't infer the element type, then let schema evolution add it when some data comes in.